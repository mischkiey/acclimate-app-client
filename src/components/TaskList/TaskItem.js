import React from 'react';

function TaskItem(props) {
    return (
        <>  
            <div className='list-group'>
                <button
                    className='y-btn'
                    onClick={(e) => props.handleCheckTask(e, props.user_id, props.user_task_id, props.user_task_completed)}
                >
                    <i className="material-icons">check</i>
                </button>
                <input
                    className=
                        {
                            (props.user_task_completed)
                                ? 'completed item-input' 
                                : 'item-input'
                        }
                    defaultValue={props.user_task}
                    onChange={(e) => {props.handleEditTask(e, props.user_id, props.user_task_id)}}
                    type='text'
                />
                <button
                    className='r-btn'
                    onClick={() => props.handleDeleteTask(props.user_task_id)}
                >
                    <i className="material-icons">delete</i>
                </button>
            </div>    
        </>
    );

};

export default TaskItem;