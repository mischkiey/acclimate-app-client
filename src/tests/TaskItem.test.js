import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import TaskItem from '../components/TaskList/TaskItem';

describe(`TaskItem Component`, () => {
    const item = {
        user_task: 'Bring pets inside'
    };

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>   
                    <TaskItem {...item} />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <TaskItem {...item} />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})