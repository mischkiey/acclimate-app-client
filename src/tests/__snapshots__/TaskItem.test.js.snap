// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskItem Component Snapshot test Renders the UI as expected 1`] = `
<div
  className="list-group"
>
  <input
    className="item-input"
    defaultValue="Bring pets inside"
    onChange={[Function]}
    type="text"
  />
  <button
    className="r-btn"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      delete
    </i>
  </button>
</div>
`;
